gdal/port/cpl_conv.cpp

```diff
@@ -651,6 +651,36 @@ const char *CPLReadLineL(VSILFILE *fp) { return CPLReadLine2L(fp, -1, nullptr);
 const char *CPLReadLine2L( VSILFILE *fp, int nMaxCars,
                            CPL_UNUSED CSLConstList papszOptions )
 
+{
+    int nBufLength;
+    return CPLReadLine3L( fp, nMaxCars,
+                          papszOptions,
+                          &nBufLength );
+}
+
+/************************************************************************/
+/*                           CPLReadLine3L()                            */
+/************************************************************************/
+
+/**
+ * Simplified line reading from text file.
+ *
+ * Similar to CPLReadLine(), but reading from a large file API handle.
+ *
+ * @param fp file pointer opened with VSIFOpenL().
+ * @param nMaxCars  maximum number of characters allowed, or -1 for no limit.
+ * @param papszOptions NULL-terminated array of options. Unused for now.
+ * @param nBufLength[out] size of output string
```

Comment:
Should be named pnBufLength to follow naming conventions in GDAL. I'd add in he doc that the pointer must be non-NULL.

Reply:
The papszOptions argument is generally the last one in the parameter list in similar APIs